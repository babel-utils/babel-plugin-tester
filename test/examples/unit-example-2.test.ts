/* eslint jest/require-hook: ["error", { "allowedFunctionCalls": ["pluginTester"] }] */

import { pluginTester } from 'universe';

import { identifierReversePlugin } from 'testverse:helpers/plugins.ts';

// * Simple example from README.md
pluginTester({
  plugin: identifierReversePlugin,
  snapshot: true,
  tests: [
    {
      code: '"hello";'
      // Snapshot should show that code has not changed.
    },
    {
      snapshot: false,
      code: 'var hello = "hi";',
      output: "var olleh = 'hi';"
    },
    `
      function sayHi(person) {
        return 'Hello ' + person + '!'
      }
      console.log(sayHi('Jenny'))
    `
  ]
});
