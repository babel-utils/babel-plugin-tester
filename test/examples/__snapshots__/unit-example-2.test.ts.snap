// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

"hello";

      ↓ ↓ ↓ ↓ ↓ ↓

'hello';

`;

exports[`unknown plugin 3. unknown plugin: 3. unknown plugin 1`] = `


function sayHi(person) {
  return 'Hello ' + person + '!'
}
console.log(sayHi('Jenny'))
    

      ↓ ↓ ↓ ↓ ↓ ↓

function iHyas(nosrep) {
  return 'Hello ' + nosrep + '!';
}
elosnoc.gol(iHyas('Jenny'));

`;
